unstructured/ingest/cli/cmds/pinecone.py

```diff
@@ -0,0 +1,66 @@
+import typing as t
+from dataclasses import dataclass
+
+import click
+
+from unstructured.ingest.cli.interfaces import (
+    CliMixin,
+)
+from unstructured.ingest.interfaces import BaseConfig
+
+
+@dataclass
+class PineconeCliWriteConfig(BaseConfig, CliMixin):
```

Comment:
Since this is actually extending configs from the Pinecone Connector file, I've been importing it from there and extending them directory: 
```suggestion
from unstructured.ingest.connector.pinecone import SimplePineconeConfig
class PineconeCliWriteConfig(SimplePineconeConfig, CliMixin):
...
```

Reply:
This didn't work, giving:
`Error: type object 'PineconeCliWriteConfig' has no attribute 'from_dict'`
when I ran the test

Reply:
Might be because SimplePineconeConfig implements session handle creation, that's the only difference I can see with the MongoDB configs

Reply:
You need this to also be decorated with `@dataclass`

Reply:
It was already decorated, but I've found why it happens now:
- the connector config does not inherit from `BaseConfig`, and we need a descendant of `BaseConfig` to provide `BaseConfig`'s functionality
- so I've replaced `CliMixin` with `CliConfig` (which inherits from both `CliMixin` and `BaseConfig`, therefore providing the needed `BaseConfig` functionality)

addressed with [update parent classes for cliconfig](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/a2b2fc379f38280c800a9215ecc8da2ca5400a90)